Nmerey/rsp
# RPS Feature

### PR Description: Implement Rock-Paper-Scissors Game

#### Summary
This pull request introduces the core functionality for a Rock-Paper-Scissors game. The implementation includes the `GamesController` with actions to handle game logic and the corresponding unit tests to ensure the correctness of the game mechanics.

#### Changes
1. **GamesController**:
   - Added `index` action (currently empty).
   - Added `create` action to handle the game logic:
     - Fetches the user's throw from the request parameters.
     - Fetches the server's throw using the `FetchCurbThrowService`.
     - If the server's throw is `nil`, a random throw is selected from the predefined `WinningCombinations::THROW_SIGNS`.
     - Determines the game result using the `RpsResultService`.
     - Returns the result and the server's throw as a JSON response.

2. **Unit Tests**:
   - Added unit tests for the `GamesController#create` action:
     - Ensures a successful response is returned.
     - Verifies the correct result when the server throw is fetched successfully.
     - Verifies the correct result when the server throw is `nil`.
     - Handles scenarios with invalid user throws.
     - Handles draw scenarios where both user and server throws are the same.

#### How to Test
1. Run the Rails server and navigate to the appropriate endpoint to test the game manually.
2. Execute the unit tests using RSpec to ensure all tests pass:
   ```sh
   bundle exec rspec
   ```
### Screenshots

![Screenshot from 2024-06-13 11-57-56](https://github.com/Nmerey/rsp/assets/32390632/0f875f5e-c091-4c4b-a9eb-1c2d1541941a)
![Screenshot from 2024-06-13 11-57-43](https://github.com/Nmerey/rsp/assets/32390632/0e440a66-9100-44c7-b2b2-772472498623)

#### Notes
- The `index` action is currently a placeholder and can be expanded in future iterations.
- The `FetchCurbThrowService` is stubbed in the tests to control the server throw values for predictable test outcomes.

This PR sets the foundation for the Rock-Paper-Scissors game and ensures that the core game logic is functioning as expected. Further enhancements and features can be built upon this base.
